### PR TITLE
Improve serializer typing

### LINKED ISSUE #9: Conflict between typing constang SERIALIZABLE and dynamic possibilities of serializer factory.
A line like `SERIALIZABLE = Union[ser_factory.get_serialazables()]` and later using `some_parameter: SERIALIZABLE` does not work:
A static type checker (like mypy) does not execute the code and so does not know, which objects are serializable.

Do I understand it correctly, that the PyMoDAQ objects subclass `SerializableBase`?
In this case, we can define `SERIALIZABLE = Union[BUILTIN_SERIALIZABLE, SerializableBase]`, where we list all builtins, which are manually added.

That makes all classes, which implement the functions defined in `SerializableBase` automatically fit for serialization, even for mypy.

### PR DESCRIPTION
**Pull request by BenediktBurger**
Closes #9

### CODE REVIEW

File: src/pymodaq_data/serialize/factory.py [hunk 1]
```diff
@@ -19,21 +21,31 @@ def type(cls):
...
 
 
+# List of all objects serializable via the serializer
+SERIALIZABLE = Union[bytes, str, int, float, complex, list, NDArray, SerializableBase]
+
+Serializable = TypeVar("Serializable", bound=SERIALIZABLE)
+_SerializableClass = TypeVar("_SerializableClass", bound=SerializableBase)
+
+Serializer = Callable[[Serializable], bytes]
+Deserilizer = Callable[[bytes], Tuple[Serializable, bytes]]
```
Comment (seb5g) [inline src/pymodaq_data/serialize/factory.py:42]: call it Deserializer

**Review by seb5g — CHANGES_REQUESTED** [review]
you're good at typing ;-) Please just modify the typo in deserializer name.

File: src/pymodaq_data/serialize/factory.py (hunk 1 above)
Comment (seb5g) [reply src/pymodaq_data/serialize/factory.py:42]: actually I'm on it so I will accept the PR and modify the name myself

Comment (BenediktBurger) [reply src/pymodaq_data/serialize/factory.py:42]: Oh. Thanks

**Pull request merged**